src/pkg/caendr/caendr/utils/bio.py

```diff
@@ -16,3 +19,62 @@ def arm_or_center(chrom, pos):
   if pos > ca[2]:
       c = 'arm'
   return c
+
+
+
+def parse_chrom_interval(interval, silent=True):
+  '''
+    Parse a string representing a chromosome interval into a dict.
+
+    Args:
+      interval (str): The interval string to parse.
+      silent (bool, optional): If False, throw a ValueError if the given value is not a valid interval. If True, returns None instead. Default True.
+
+    Returns:
+      dict:
+        - chrom (str): The chromosome
+        - start (int): The start location
+        - stop (int):  The stop location
+  '''
+
+  # Perform RegEx search on interval string
+  match = re.search(CHROM_INTERVAL_REGEX, interval.replace(',',''))
+
+  # If there was a match, parse the capture groups
+  if match:
+    return {
+      'chrom': match.group(1),
+      'start': int(match.group(2)),
+      'stop':  int(match.group(3)),
+    }
+
+  if not silent:
```

Comment:
hum...we should review the behavior for `silent==True`, should it log something?

Reply:
I'm inclined to say that if we want a log message, the caller should probably be using `silent=False` and logging the error itself.  That said, I can see why we might want to add a log message anyways, in case the caller doesn't do that.  Happy to add one if you think it would be useful.

--

A bit more broadly -- my idea for the parameter was to have a nice way to toggle between "just return `None`" and "raise an error" in case the pattern doesn't match.  Then, it would be up to the caller to determine which behavior is more useful.  Implicit in that would be that `silent=True` would indicate that a failed match isn't a reason to halt execution.

E.g. at the top level of the `variant_annotation.query_interval` view function, we return an empty response if the result is `None`, whereas in `run_interval_query`, the whole query should fail with the appropriate error if the interval can't be parsed.  That said, I can also see an argument for raising an error in both cases, and for catching it in the former case & letting it propagate in the latter.

Reply:
Got it. If one squints, it might feel like we are trying to implement workaround a throw/catch. Would it be reasonable to expect the function to throw (in some cases) and let the caller handle it with try/catch? At that point the caller can decide how it wants to proceed in case of an error. 

I'm not opposed to the extra `silent` argument in this functions; I am mainly looking for ways to simplify and reduce cognitive load throughout the codebase & function signatures.

lmk what you think.

Reply:
Looking back at how we use these functions, I think throw/catch like you suggested does make more sense.  Updated in https://github.com/AndersenLab/CAENDR/pull/373/commits/551b3abf14777bcf657e4a39be576e4b487d5101.